benchmarking/src/time_bench.rs

```diff
@@ -0,0 +1,322 @@
+    #[arg(long, short)]
+    pub time_limit: Option<humantime::Duration>,
+    /// Output in JSON rather than a human-readable format
+    #[arg(long)]
+    pub json: bool,
+    /// Program to run
+    pub prog: OsString,
+    /// Arguments to pass
+    pub args: Vec<OsString>,
+}
+
+/// Get the temporal-verifier root path.
+/// TODO: Only works when using a binary compiled locally. Will need to fix when
+/// distributing a precompiled binary.
+#[allow(non_snake_case)]
+pub fn REPO_ROOT_PATH() -> &'static Path {
+    Path::new(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .expect("could not get parent directory of benchmarking package")
+}
+
+fn time_bin() -> PathBuf {
+    let target = REPO_ROOT_PATH().join("target");
+    let release = target.join("release/time");
+    if release.exists() {
+        return release;
+    }
+    let debug = target.join("debug/time");
+    if debug.exists() {
+        return debug;
+    }
+    panic!("time binary not available - try cargo build --bin time")
+}
+
+/// Run cargo and print a warning if it fails
+fn run_cargo<S, I>(args: I)
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let args = args
+        .into_iter()
+        .map(|s| s.as_ref().to_owned())
+        .collect::<Vec<_>>();
+    let status = Command::new("cargo")
+        .arg("--quiet")
+        .args(&args)
+        .stdin(Stdio::null())
+        .stdout(Stdio::null())
+        .stderr(Stdio::inherit())
+        .current_dir(REPO_ROOT_PATH())
+        .status();
+    match status {
+        Ok(status) => {
+            if !status.success() {
+                eprintln!(
+                    "cargo {} failed",
+                    args.join(OsStr::new(" ")).to_string_lossy()
+                );
+            }
+        }
+        Err(err) => {
+            eprintln!("could not run cargo: {err}");
+        }
+    }
+}
+
+/// Build the `time` binary, which is an implicit dependency of benchmarks.
+pub fn compile_time_bin() {
+    run_cargo(["build", "--release", "--bin", "time"]);
+}
+
+/// Compile the `temporal-verifier` binary, which is run by the benchmarks.
+pub fn compile_flyvy_bin() {
+    run_cargo(["build", "--release", "--bin", "temporal-verifier"]);
+}
+
+#[derive(Debug, Clone)]
+struct RawRunMeasurement {
+    real_time: Duration,
+    usage: Usage,
+    self_usage: Usage,
+    timed_out: bool,
+    status: Option<i32>,
+}
+
+fn time_val_to_duration(time: TimeVal) -> Duration {
+    Duration::from_secs(time.tv_sec() as u64) + Duration::from_micros(time.tv_usec() as u64)
+}
+
+impl RawRunMeasurement {
+    fn into_measurement(self) -> RunMeasurement {
+        RunMeasurement {
+            real_time: self.real_time,
+            user_time: time_val_to_duration(self.usage.user_time()),
+            sys_time: time_val_to_duration(self.usage.system_time()),
+            self_user_time: time_val_to_duration(self.self_usage.user_time()),
+            max_mem_bytes: self.usage.max_rss() as usize,
```

Comment:
I looked into linux versus mac a little bit and the first thing I noticed is that the memory usage on linux is not reported correctly. Apparently this is because `getrusage` on linux returns max RSS in *kilobytes* while `getrusage` on macOS returns max RSS in *bytes*. So we need to conditionally multiply by 1024 on linux to get the number in bytes.

Reply:
Wow that's really lame. Thanks for finding this! Will fix.

Reply:
Finally fixed this.